tests/unit/deploy/test_newdeployer.py

```diff
@@ -0,0 +1,692 @@
+            'f': {'name': 'appname-dev-f',
+                  'autogenerated': False,
+                  'policy_file': 'policy-dev.json'},
+            'g': {'name': 'appname-dev-g',
+                  'autogenerated': False,
+                  'policy_file': 'g.json'},
+            'h': {'name': 'appname-dev-h',
+                  'autogenerated': False,
+                  'policy_file': 'h.json'},
+        }),
+]
+
+
+@pytest.mark.parametrize('case', ROLE_TEST_CASES)
+def test_role_creation(case):
+    _, config = case.build()
+    builder = ResourceGraphBuilder()
+    application = builder.build(config, stage_name='dev')
+    case.assert_required_roles_created(application)
+
+
+class TestDefaultsInjector(object):
+    def test_inject_when_values_are_none(self):
+        injector = InjectDefaults(
+            lambda_timeout=100,
+            lambda_memory_size=512,
+        )
+        function = models.LambdaFunction(
+            # The timeout/memory_size are set to
+            # None, so the injector should fill them
+            # in the with the default values above.
+            timeout=None,
+            memory_size=None,
+            resource_name='foo',
+            function_name='app-dev-foo',
+            environment_variables={},
+            runtime='python2.7',
+            handler='app.app',
+            tags={},
+            deployment_package=None,
+            role=None,
+        )
+        config = Config.create()
+        injector.handle(config, function)
+        assert function.timeout == 100
+        assert function.memory_size == 512
+
+    def test_no_injection_when_values_are_set(self):
+        injector = InjectDefaults(
+            lambda_timeout=100,
+            lambda_memory_size=512,
+        )
+        function = models.LambdaFunction(
+            # The timeout/memory_size are set to
+            # None, so the injector should fill them
+            # in the with the default values above.
+            timeout=1,
+            memory_size=1,
+            resource_name='foo',
+            function_name='app-stage-foo',
+            environment_variables={},
+            runtime='python2.7',
+            handler='app.app',
+            tags={},
+            deployment_package=None,
+            role=None,
+        )
+        config = Config.create()
+        injector.handle(config, function)
+        assert function.timeout == 1
+        assert function.memory_size == 1
+
+
+class TestPolicyGeneratorStage(object):
+    def test_invokes_policy_generator(self):
+        generator = mock.Mock(spec=AppPolicyGenerator)
+        generator.generate_policy.return_value = {'policy': 'doc'}
+        policy = models.AutoGenIAMPolicy(models.DeployPhase.BUILD)
+        config = Config.create()
+
+        p = PolicyGenerator(generator)
+        p.handle(config, policy)
+
+        assert policy.document == {'policy': 'doc'}
+
+    def test_no_policy_generated_if_exists(self):
```

Comment:
Maybe something more descriptive than `exists`, like `_if_xxx_populated` which was used in a test below.